h2o-core/src/main/java/water/api/schemas3/FrameV3.java

```diff
@@ -261,11 +284,18 @@ private FrameV3 fillFromImpl(Frame f, long row_offset, int row_count,
         Log.warn("For Frame: " + f._key + ", Vec number: " + (column_offset + i) + " (" + f.name((column_offset + i))+ ") is missing; not returning it.");
       else
         vecs[column_offset + i].startRollupStats(fs);
+
+
     for( int i = 0; i < column_count; i++ )
       if (null == DKV.get(vecs[column_offset + i]._key))
         Log.warn("For Frame: " + f._key + ", Vec number: " + (column_offset + i) + " (" + f.name((column_offset + i))+ ") is missing; not returning it.");
-      else
-        columns[i] = new ColV3(f._names[column_offset + i], vecs[column_offset + i], this.row_offset, this.row_count);
+      else {
+        if(i < full_column_count) {
```

Comment:
This can be simplified - the condition can be in the last argument, 'if' is not necessary

Reply:
Good point, thanks!